src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,180 +1,165 @@
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useState, ChangeEvent, useMemo } from "react";
+import classNames from "classnames";
+
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
 import { VenueEvent } from "types/venues";
 import { Room, RoomTypes } from "types/rooms";
-import { User } from "types/User";
 
-import { WithId } from "utils/id";
-import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 import { isTruthy } from "utils/types";
+import { uppercaseFirstChar } from "utils/string";
+import { formatUtcSecondsRelativeToToday } from "utils/time";
+import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 
 import { useWorldUsers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
 import { RoomModal } from "components/templates/PartyMap/components";
 
-import "./NavSearchBar.scss";
 import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+import { NavSearchResult } from "./NavSearchResult";
+
+import "./NavSearchBar.scss";
+
+const emptyEventsArray: VenueEvent[] = [];
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
   const onSearchInputChange = useCallback(
-    (e) => setSearchQuery(e.target.value),
+    (e: ChangeEvent<HTMLInputElement>) =>
+      setSearchQuery(e.target.value.toLowerCase()),
     []
   );
```

Comment:
[NB] We might want to use a debounce/throttle here so that it doesn't update on every keystroke?

- https://lodash.com/docs/4.17.15#debounce
- https://lodash.com/docs/4.17.15#throttle

Reply:
applied debounce here: https://github.com/sparkletown/sparkle/pull/1328/commits/699c5a3ca1a555eed71de9b23fb0ba94cbe172ab